src/components/ComboButton/ComboButton.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file Combo button.
+ * @copyright IBM Security 2019
+ */
+
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useRef, useState } from 'react';
+import ChevronDown16 from '@carbon/icons-react/lib/chevron--down/16';
+import ChevronUp16 from '@carbon/icons-react/lib/chevron--up/16';
+import { settings } from 'carbon-components';
+
+import Button from '../Button';
+import OverflowMenu from '../OverflowMenu';
+import OverflowMenuItem from '../OverflowMenuItem';
+import { TooltipDirection } from '../IconButton/IconButton';
+
+import { namespace as buttonNamespace } from '../Button/Button';
+
+import { getComponentNamespace } from '../../globals/namespace';
+
+export const namespace = getComponentNamespace('combo-button');
+
+const { prefix } = settings;
+
+const ComboButton = ({ children, className, direction }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const wrapper = useRef(null);
+
+  const getMenuOffset = () => {
+    const { top } = wrapper.current.getBoundingClientRect();
+    const isTop = direction === TooltipDirection.TOP;
+    return {
+      top: isTop ? top : top * -1,
+      left: 'auto',
+    };
+  };
+
+  const childrenArray = React.Children.toArray(children);
+
+  // Save first child (e.g., primary action) to use as a `Button`:
+  const primaryAction = React.cloneElement(childrenArray[0]);
+  const primaryActionWithProps = React.Children.toArray(primaryAction).map(
+    button => {
+      const { children, ...rest } = button.props;
+      return (
+        <Button
+          {...rest}
+          key={`primary-action-button-${button.id}`}
+          className={`${namespace}--primary`}
+        >
+          {React.Children.toArray(children)}
+        </Button>
+      );
+    }
+  );
+
+  // Save remaining children to be displayed in the `OverflowMenu`:
+  let overflowItems;
+  let overflowMenuItemWithProps;
+  if (childrenArray.length > 1) {
+    overflowItems = childrenArray.slice(1);
+
+    // Create `OverflowMenuItem` components:
+    overflowMenuItemWithProps = React.Children.toArray(overflowItems).map(
```

Comment:
I don't think you need to run this through `toArray` again since `overflowItems` is already an array